swap_meet/vendor.py

```diff
@@ -1,2 +1,113 @@
 class Vendor:
-    pass
\ No newline at end of file
+    """
+    A class to represent a vendor, managing the inventory and providing
+    methods for adding, removing, and swapping items between vendors.
+    """
+
+    def __init__(self, inventory=None):
+        """
+        Initializes a Vendor instance with an optional inventory list.
+        If no inventory is provided, an empty list is initialized.
+        """
+        self.inventory = inventory or []
+
+    def add(self, new_item_added):
+        """
+        Adds the item to the `inventory`
+        and returns item
+        """
+        self.inventory.append(new_item_added)
+        return new_item_added
+
+    def remove(self, item_to_remove):
+        """
+        Removes the specified item from the `inventory` if it exists and returns item
+        """
+        if item_to_remove not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item_to_remove)
+            return item_to_remove
+
+    def get_by_id(self, item_id):
+        """
+        Check if unique item id matches an item id in inventory and return item
+        """
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        """
+        Swaps the specified items between this vendor and another vendor,
+        if both vendors have the specified items in their inventory.
+        """
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        return False
```

Comment:
Prefer to use the guard clause pattern here by inverting logic so we check for invalid input first and immediately return False. 

Then your main logic on lines 46-50 can be unindented one level, which would emphasize that this is the main logic for the method.

```py
def swap_items(self, other_vendor, my_item, their_item): 
    if my_item not in self.inventory and their_item not in other_vendor.inventory: 
        return False 

self.inventory.remove(my_item) 
self.inventory.append(their_item) 

other_vendor.inventory.append(my_item) 
other_vendor.inventory.remove(their_item) 

return True

Reply:
Removing and appending elements to a list works! If you haven't seen it already, Python also has syntax for assigning multiple values in one line that we can leverage to swap elements in a list (read more [here](https://note.nkmk.me/en/python-multi-variables-values/))

For example:
```py
a, b = 100, 200

print(a)
# 100

print(b)
# 200
```

We can use that syntax to swap items like so:
```py
my_item_index = self.inventory.index(my_item)
their_item_index = other_vendor.inventory.index(their_item)
other_vendor.inventory[their_item_index], self.inventory[my_item_index] = my_item, their_item
```